framework/src/base/Moose.C

```diff
@@ -572,6 +575,7 @@ bool _warnings_are_errors = false;
 bool _deprecated_is_error = false;
 bool _throw_on_error = false;
 bool _throw_on_warning = false;
+int autosave_flag = 0;
```

Comment:
Have you tried making this a bool? Currently you have a signed int but you are calling it a flag. You are using it as a Boolean in your class. 

After making the comment above we probably want an integer type (probably unsigned) here, but you you might not call it a flag (misnomer) in that case. This would be handy to handle multiple different signals at the same time (bit mask). I won't require that for now, at the very least though put the integer at the top or the bottom (good practice). The compiler can potentially lump all these bools together but the int will cause an alignment issue.

Reply:
We discussed making it an int so that we can expand it to include checks for different kinds of signals in later development